crates/primitives/src/alloy_compat.rs

```diff
@@ -227,7 +233,31 @@ impl TryFrom<alloy_rpc_types::Transaction> for Transaction {
                 }))
             }
             #[cfg(feature = "optimism")]
-            Some(TxType::Deposit) => todo!(),
+            Some(TxType::Deposit) => {
+                Ok(Self::Deposit(TxDeposit {
+                    // TODO: Uses the wrong error here, need alloy ConversionError enum updated with new errors for this.
+                    source_hash: match &tx.other["sourceHash"] {
```

Comment:
otherfields has `get_deserialized::<String>(key)` which we should use combined with `Option::transpose`

Reply:
thanks for the tip. what's the reasoning to use `Option::transpose` since i'm going to have to unwrap to the String anyways in order to parse into B256?

i would think ideal would be something like:

```rust
source_hash: tx.other.get_deserialized::<String>("sourceHash")
    .ok_or(ConversionError::MissingSourceHash)?
    .map_err(|_| ConversionError::InvalidSourceHash)?
    .parse()
    .map_err(|_| ConversionError::InvalidSourceHash)?,
```

Reply:
agreed that `Option::transpose` better for deserializing mint, since we want to deal with the error and then match on the option, something like:
```rust
mint: match Option::transpose(tx.other.get_deserialized::<u128>("mint"))
    .map_err(|_| ConversionError::MissingBlockNumber)? {
        Some(mint) => if mint == 0 { None } else { Some(mint) },
        None => None,
    },
```
do you agree with both of these or see a cleaner way to do it?